examples/rapids/02-rapids-gpu-cluster.ipynb

```diff
@@ -0,0 +1,542 @@
+    "\n",
+    "    df = df[df[\"fare_amount\"] > 0]  # to avoid a divide by zero error\n",
+    "    df[\"tip_fraction\"] = df[\"tip_amount\"] / df[\"fare_amount\"]\n",
+    "    df[\"target\"] = df[\"tip_fraction\"] > 0.2\n",
+    "\n",
+    "    df[\"pickup_weekday\"] = df[\"tpep_pickup_datetime\"].dt.weekday\n",
+    "    df[\"pickup_hour\"] = df[\"tpep_pickup_datetime\"].dt.hour\n",
+    "    df[\"pickup_week_hour\"] = (df[\"pickup_weekday\"] * 24) + df.pickup_hour\n",
+    "    df[\"pickup_minute\"] = df[\"tpep_pickup_datetime\"].dt.minute\n",
+    "\n",
+    "    df = df[\n",
+    "        [\n",
+    "            \"pickup_weekday\",\n",
+    "            \"pickup_hour\",\n",
+    "            \"pickup_week_hour\",\n",
+    "            \"pickup_minute\",\n",
+    "            \"passenger_count\",\n",
+    "            \"PULocationID\",\n",
+    "            \"DOLocationID\",\n",
+    "            \"target\",\n",
+    "        ]\n",
+    "    ]\n",
+    "\n",
+    "    df = df.astype(\"float32\").fillna(-1)\n",
+    "    df[\"target\"] = df[\"target\"].astype(\"int32\")\n",
+    "\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<br>\n",
+    "<details><summary>If you want to learn more about how this function works, click here. </summary>\n",
+    "    \n",
+    "* **Creates a target column** - The raw data does not contain a column that cleanly describes whether a particular ride could be considered a \"good\" tip. So we need to add one! In this exercise, a \"good\" tip is one that is greater than 20% of the fare amount. This function makes this a binary classification problem.\n",
+    "* **Adds additional features** - Here, instead of giving a model a raw timestamp, we provide multiple derived characteristics like hour of the day and day of the week. Giving the machine learning model a richer description of each training observation improves its ability to describe the relationship between those observations' characteristics and the target.\n",
+    "* **Removes Unused Columns** - If the goal is to produce a model that could predict whether a tip will be good *at the beginning of the trip*, then characteristics that can only be known **AFTER** the tip must be excluded. For example, you can't know the dropoff time or the type of payment until a ride has concluded.\n",
+    "* **Cast the Columns to 32-bit types** - This reduces the memory footprint of the model and insures compatablility with older version of `cuml`.\n",
+    "    \n",
+    "</details>\n",
+    "<br> \n",
+    "We now just need to run the raw data through our function to prepare it for machine learning."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "taxi = prep_df(taxi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***\n",
```

Comment:
Again I would take the lines out. It's just not clear what they are trying to indicate and if you are suggesting a topic change you could use a header to do so